### PR TITLE
improve docker build

### PR DESCRIPTION
**Pull request by saada**
Improve docker build performance from 6m to 2m (no caching) using docker buildx job. It also includes state of the art docker caching

### CODE REVIEW

**Review by paulcarlton-ww — APPROVED** [review]

**Comment by saada** [comment]
Ready for review

**Review by paulcarlton-ww — DISMISSED** [review]
Maybe update dev guide and cover helm chart and image versions as well as integration tests too?

File: pkg/apply/layerApplier.go [hunk 1]
```diff
@@ -114,7 +114,7 @@ func getLabel(hr metav1.ObjectMeta) string {
 func removeResourceVersion(obj runtime.Object) string {
```
Comment (paulcarlton-ww) [inline pkg/apply/layerApplier.go:114]: see https://github.com/fidelity/kraan/pull/184/files#diff-9126395a3abfddd5d452095ceff4984434148f8370530962de3aef92d4b08202L114 I fixed this by removing the returned value, it is not actually used

**Review by paulcarlton-ww — COMMENTED** [review]
Can you change removeResourceVersion in ApplierLayers.go in line with my change in #184 please

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -10,66 +10,69 @@ on:
...
+          go-version: 1.15
+      - name: Setup dependencies
+        run: make setup
+      - name: Lint tasks
         run: |
           make validate-versions
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.37.1
```
Comment (paulcarlton-ww) [inline .github/workflows/main.yaml:32]: if we change the golangci-lint version we need to update bin/setup.sh https://github.com/fidelity/kraan/blob/e23e382980e25408e5f9412fba1e63d4858dd885/bin/setup.sh#L3 and docs/dev-guide.md too

**Review by wangli1030 — APPROVED** [review]

**Pull request merged**